src/views/board_view/mod.rs

```diff
@@ -144,35 +141,35 @@ impl BoardView {
         self.set_state(state);
     }
 
-    fn handle_mouse_event(&mut self, event: MouseEvent) {
+    fn handle_pointer_event(&mut self, event: PointerEvent) -> bool {
+        // TODO(Menno 06.08.2025) Hightlight pieces if we hover over them
+        let mut handled = false;
         match event {
-            MouseEvent::Down(coordinates) => {
-                self.mouse_is_down = true;
+            PointerEvent::Down(coordinates) => {
```

Comment:
Note that there'll be a pointerId associated with these events, which identifies the finger doing the dragging. Ideally you want to keep track of this id on the first pointerDown, and ignore all further events that don't have the same id. This prevents errant fingers (e.g. from children) from double-starting drags.